### ISSUE TITLE
Issues with Typescript and Standalone Commands

### ISSUE THREAD

**Issue by danfsd** [tj/commander.js#1259]
Hello, I'm trying to convert my project **with standalone commands to Typescript** but I'm encountering the following problems:

1. VSCode is not offering tab completions from the included typings. I even tried to install `@types/commander` as a devDependency but the issue still persists. See my `tsconfig.json` file in the next section. (note: I also have `@types/node` installed as a dev dependency)
2. I'm receiving the following Typescript warning when trying to require `program` in the top of my standalone command files: `Cannot redeclare block-scoped variable 'program'`. I know that this is not a bug but rather is a feature from Typescript. The thing is that the way standalone commands works conflicts with the way Typescript understands the scopes.

Along with the fixes or workarounds for the questions above, I would like to ask if there are any other tips for getting Commander to work with Typescript when using Standalone commands.

Here's my **tsconfig.json** file:
```
{
    "compilerOptions": {
        "target": "ES6",
        "module": "CommonJS",
        "lib": ["ES6", "ES2015", "DOM"],
        "declaration": true,
        "outDir": "lib",
        "rootDir": "src",
        "strict": false,
        "types": ["node"],
        "esModuleInterop": true,
        "resolveJsonModule": true,        
    }
}
```

**Comment by danfsd** [comment]
Hey guys, just discovered that problem 1. and 2. were solved by switching from `require` to `import`.

Nevertheless, those are still issues (at least problem 2. is) if one does want to make things work using `require`.

**Comment by shadowspawn** [comment]
It is expected that you will use `import` to get the TypeScript typings. `require` is for plain old JavaScript modules.

I am not able to reproduce the problems in a new project using your `tsconfig.json`, using either `import` or `require`. What is some code that shows the problem? 

I did get the error when I accidentally declared `program` twice, but expected in that case!

**Comment by danfsd** [comment]
**TL;DR**
Everything is working fine now that I have switched to `import` instead of `require`.

**Explanation**
> It is expected that you will use import to get the TypeScript typings. require is for plain old JavaScript modules.

Gotcha, its really working fine with the `import`, so it's not an issue anymore.

I think that the other issue regarding the redeclaration of the `program` variable was due to the usage of the `require` instead of `import`, which requires that we assign the required module to a `var`, `let` or `const`. As I am importing the modules now, I don't have to assign the module it to anything thus the problem also was solved.

But just to clarify what was happening, please see below

```
const { program } = require('commander');

program
  .command('test', 'description')
  .parse(process.argv);
```

index-test.ts
```
// Here, the editor was showing that warning I mentioned before
const { program } = require('commander');
// ...
```

The workaround (explained [here](https://medium.com/@muravitskiy.mail/cannot-redeclare-block-scoped-variable-varname-how-to-fix-b1c3d9cc8206)) I had to do in order for the warning to go away was to place an export statement on the very top of the `index-test.ts` file like so:

```
export {};
// The editor didn't complained about the redeclaration anymore
const { program } = require('commander');
// ...

**Comment by shadowspawn** [comment]
Thanks for additional info, with *two* files I can now reproduce. :-)

Thanks for explanation and work-around. Not related to commander as such, but multiple files with same named variables in global "script" scope.
